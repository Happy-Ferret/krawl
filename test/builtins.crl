import "stdio.h";

type Dummy struct {
	a, b, c int;
	d *Dummy;
	e double;
};

func main(argc int, argv **byte) int
{
	var d Dummy;

	stdio.printf("sizeof(type Dummy) == %d\n", sizeof(type Dummy));
	stdio.printf("sizeof(type [3]Dummy) == %d\n", sizeof(type [3]Dummy));
	stdio.printf("sizeof(d) == %d\n", sizeof(d));
	stdio.printf("sizeof(&d) == %d\n", sizeof(&d));
	return 0;
}
