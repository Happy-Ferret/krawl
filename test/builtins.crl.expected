sizeof(type Dummy) == 24
sizeof(type [3]Dummy) == 72
sizeof(d) == 24
sizeof(&d) == 4
